### PR TITLE
[Yargs] Allow Command modules to be passed into command fn

### PR DESCRIPTION
**Pull request by jmalonzo**
Please fill in this template.

- [x] Prefer to make your PR against the `types-2.0` branch.
- [ ] The package does not provide its own types, and you can not add them.
- [x] Test the change in your own code.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).

If adding a new definition:
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Run `tsc` without errors.
- [ ] Include the required [files](https://github.com/DefinitelyTyped/DefinitelyTyped#create-a-new-package) and header.

If changing an existing definition:
- [x] Provide a URL to  documentation or source code which provides context for the suggested changes: <http://yargs.js.org/docs/#methods-commandmodule>
- [ ] Increase the version number in the header if appropriate.

### CODE REVIEW

File: yargs/index.d.ts [hunk 1]
```diff
@@ -199,6 +200,14 @@ declare namespace yargs {
         nargs?: number;
     }
 
+    interface CommandModule {
+        aliases?: Array<string>|string;
```
Comment (pushplay) [inline yargs/index.d.ts]: As a matter of style use `string[]` instead of `Array<string>`

File: yargs/index.d.ts [hunk 2]
```diff
@@ -199,6 +200,14 @@ declare namespace yargs {
...
+        builder: {[key: string]: Options} | ((args: Argv) => Argv);
+        command: Array<string>|string;
```
Comment (pushplay) [inline yargs/index.d.ts]: Same.

**Review by pushplay — CHANGES_REQUESTED** [review]
Minor changes requested.

**Comment by jmalonzo** [comment]
@pushplay Done. Thanks!

**Comment by pushplay** [comment]
Awesome.  Thanks.

👍

**Comment by poelstra** [comment]
@jmalonzo Nice.

It seems the docs allow more than the current typings:
- `.command` and `.describe` seem to be optional ("Or if the module does not export command and describe (or if you just want to override them)")
- It isn't documented, but the code seems to suggest that `.builder` is optional, too
- `command()` has a version that allows to override command and(/or?) description

Can you please update the PR?

**Comment by jmalonzo** [comment]
@poelstra Ah, yep. I think I've made the changes you requested. Can you please check again? Tnx

**Comment by poelstra** [comment]
@jmalonzo looks great, thnx!

:+1:

**Pull request merged**